### PR TITLE
#1 - v0.1.0 Candidate

### PR DESCRIPTION
**Pull request by Mikeultron**
### Changelog
- Refactoring project structure
- Configuring storybook
- Create button
- Add babel configuration
- Utilities

### CODE REVIEW

File: src/components/atoms/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { IButtonProps } from "../../../utils/interfaces";
+import { PrimaryButton, SecondaryButton, GhostButton } from "./styles";
+
+function Button({ theme, buttonStyle, children, ...rest }: IButtonProps) {
+  const renderButton = () => {
+    switch (buttonStyle) {
+      case "primary":
```
Comment (prawito) [inline src/components/atoms/Button/index.tsx:8]: You can remove case "primary" if you use the default Primary button as a default

File: src/components/atoms/Button/styles.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export const PrimaryButton = styled.button`
+  background-color: ${(props) => props.theme.background ?? colors.primary[100]};
+  color: ${(props) => props.theme.foreground ?? "white"};
+  border-radius: 999px;
+  font-family: ${fontFamily.poppins.regular};
+  outline: none;
+  border: none;
+  cursor: pointer;
+  padding: 10px 20px;
```
Comment (prawito) [inline src/components/atoms/Button/styles.ts:12]: Please use the spacing function to generate space number

File: src/utils/interfaces.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { ButtonHTMLAttributes } from "react";
+import { ButtonType } from "./types";
+
+export interface IButtonTheme {
+  background?: string;
+  foreground?: string;
+}
+
+export interface IButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  buttonStyle?: ButtonType;
```
Comment (prawito) [inline src/utils/interfaces.ts:10]: Change to variant

File: src/utils/styles.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+export const fontSize = {
+  XL: "1rem",
+  L: "0.6666666666666667rem",
+  M: "0.5rem",
+  S: "0.3888888888888889rem",
+};
+
+export const breakPoints = {
```
Comment (prawito) [inline src/utils/styles.ts:65]: remove this

**Review by prawito — CHANGES_REQUESTED** [review]

File: src/components/atoms/Button/index.tsx (hunk 1 above)
Comment (Mikeultron) [reply src/components/atoms/Button/index.tsx:8]: I changed the default case instead of the primary case for explicit type reason.

**Review by prawito — APPROVED** [review]

**Pull request merged**